### PR TITLE
For #25985: Remove ability to toggle wallpapers and code for firefox logo animation

### LINKED ISSUE #25985: Remove ability to toggle wallpapers by tapping Firefox logo
The ability to toggle the selected wallpaper by tapping the Firefox logo introduces additional constraints on the freedom to choose different wallpaper download strategies. It is also a feature that can be activated accidentally pretty easily, leading to unexpected results. It should be removed for now.

Acceptance criteria:
- Logo can no longer be tapped to switch wallpaper
- Logo no longer animates
- Settings data and preference keys used to determine if logo should animate are removed

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-20978)

### PR DESCRIPTION
**Pull request by sunilk9211**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #25985

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:49]: I'm not seeing `Flow.collect` removed anywhere in this changelog. I think we should probably revert this change to be on the safe side, unless I'm missing something.

File: app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt:41]: Generally speaking, we try to have our patches here be as small and focused as possible. Normally a small, unrelated change like this would be something we try to avoid. This is still a useful change, so I am okay with keeping it here but wanted to call it out for future reference.

**Review by MatthewTighe — CHANGES_REQUESTED** [review]
This is looking great so far, thanks again for your contribution!

There are a couple more candidates for removal. They could be done as part of this patch, or they could be done as separate follow-ups after we land this patch. I will leave that up to you. No pressure if you are not interested in them, just let me know 🙂 

1. There is a function in `WallpaperManager` called [switchToNextWallpaper](https://searchfox.org/mozilla-mobile/rev/712f60261a3cef9495814eb49148d0692c350e42/fenix/app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt#74) that was called when the logo was clicked. That should be removed, and IMO would make sense to do as part of this patch.
2. There is another user preference for whether the toggle works at all called [wallpapersSwitchedByLogoTap](https://searchfox.org/mozilla-mobile/rev/712f60261a3cef9495814eb49148d0692c350e42/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt#193). This preference and the code that uses it also need to be removed. This one would make sense as part of this patch, but I think could also be done in a follow-up.

So please let me know if you would like to add either or both of those things in this patch or in follow-ups.

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt
Comment (sunilk9211) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: > This is looking great so far, thanks again for your contribution!
> 
> There are a couple more candidates for removal. They could be done as part of this patch, or they could be done as separate follow-ups after we land this patch. I will leave that up to you. No pressure if you are not interested in them, just let me know slightly_smiling_face
> 
> 1. There is a function in `WallpaperManager` called [switchToNextWallpaper](https://searchfox.org/mozilla-mobile/rev/712f60261a3cef9495814eb49148d0692c350e42/fenix/app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt#74) that was called when the logo was clicked. That should be removed, and IMO would make sense to do as part of this patch.
> 2. There is another user preference for whether the toggle works at all called [wallpapersSwitchedByLogoTap](https://searchfox.org/mozilla-mobile/rev/712f60261a3cef9495814eb49148d0692c350e42/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt#193). This preference and the code that uses it also need to be removed. This one would make sense as part of this patch, but I think could also be done in a follow-up.
> 
> So please let me know if you would like to add either or both of those things in this patch or in follow-ups.

I have done both the changes. 
Actually i had removed switchToNextWallpaper in the first commit, but i added it back during resolving conflicts because i saw some changes related to its use, so thought will remove it once you confirm we're removing it.
Thanks for your review.

Comment (sunilk9211) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: > I'm not seeing `Flow.collect` removed anywhere in this changelog. I think we should probably revert this change to be on the safe side, unless I'm missing something.

Yes. reverted it.

File: app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt
Comment (sunilk9211) [reply app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt]: Yes i understand. Will keep in mind for future reference. Thanks. 
For now i have reverted it. At that time i wasn't able to build due to this warning.

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:756]: I think we can go ahead and fully remove this commented code

File: app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt:272]: I believe the resource this line is referenced could be removed as well.

File: app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt:77]: nit: would you add a trailing comma here? I am not sure if we have this referenced anywhere, but we have been trying to include them wherever possible. It tends to make diffs in future patches a little easier to read, as it'll reduce the number of lines highlighted.

File: app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt:265]: nit: trailing comma here as well please

File: app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt:61]: nit: another trailing comma spot 😅 apparently I missed them in the original implementation so it would be helpful to get them in now

**Review by MatthewTighe — CHANGES_REQUESTED** [review]
This is getting really close! Another thing to note is that we will want to squash these commits to make sure that our commit history is clean. Your original commit messages looks like exactly the type we'd like, so you should be able to squash down to that one. I am also happy to help with that, depending on your familiarity with git

File: app/src/main/res/values/strings.xml
Comment (MatthewTighe) [inline app/src/main/res/values/strings.xml:433]: Ah, I should've specified. We actually have a [special process](https://github.com/mozilla-mobile/fenix/wiki/Removing-strings) for removing strings. Please add these back in and tag them as described on that wiki page. Thanks for patience in getting this ship-shape!

**Review by MatthewTighe — CHANGES_REQUESTED** [review]
One last thing, then I think this should be ready to squash and land!

File: app/src/main/res/values/strings.xml
Comment (sunilk9211) [reply app/src/main/res/values/strings.xml]: Thanks for letting me know about the process. I have done the changes as mentioned there.

**Comment by hneiva** [comment]
No ci changes, removing RelEng

**Comment by sunilk9211** [comment]
Hi @MatthewTighe Sorry i had to close this PR ( https://github.com/mozilla-mobile/fenix/pull/26478 ), have opened another one. It's because the code got a little messed up in this branch.

**Comment by MatthewTighe** [comment]
No worries! Thanks for your patience in getting your branches rebased. It's always a bit of a risk when contributing to features that are being actively worked on by multiple people.

**Pull request closed without merging**
